core/graph/src/main/java/uk/gov/gchq/gaffer/operation/export/graph/handler/ExportToOtherGraphHandler.java

```diff
@@ -46,97 +47,146 @@ protected OtherGraphExporter createExporter(final ExportToOtherGraph export, fin
-        Schema schema = null;
+    private Graph getGraphAfterResolvingSchemaAndProperties(final String exportGraphId, final Schema exportSchema, final StoreProperties exportStoreProperties, final List<String> exportParentSchemaIds, final String exportParentStorePropertiesId, final GraphLibrary graphLibrary, final StoreProperties storeStoreProperties, final Schema storeSchema) {
+        final Graph rtn;
+        StoreProperties storeProperties = resolveStoreProperties(exportStoreProperties,
+                                                                 exportParentStorePropertiesId,
+                                                                 graphLibrary,
+                                                                 storeStoreProperties);
 
-        // No store graph library so we create a new Graph
-        if (null == graphLibrary) {
-            schema = null != exportSchema ? exportSchema : storeSchema;
-            final StoreProperties properties = null != exportStoreProperties ? exportStoreProperties : storeStoreProperties;
-            return new Graph.Builder()
-                    .graphId(exportGraphId)
-                    .addSchema(schema)
-                    .storeProperties(properties)
-                    .build();
-        }
+        Schema schema = resolveSchema(exportSchema,
+                                      exportParentSchemaIds,
+                                      graphLibrary,
+                                      storeSchema);
 
-        // If the graphId exists in the graphLibrary then just use it
-        if (graphLibrary.exists(exportGraphId)) {
-            return new Graph.Builder()
-                    .graphId(exportGraphId)
-                    .library(graphLibrary)
-                    .build();
-        } else {
-            if (exportParentStorePropertiesId != null) {
-                storeProperties = graphLibrary.getProperties(exportParentStorePropertiesId);
-            }
-            if (exportStoreProperties != null) {
-                if (storeProperties == null) {
-                    storeProperties = exportStoreProperties;
-                } else {
-                    // delete the old properties id as we are about to modify the properties
-                    storeProperties.getProperties().remove(ID);
-                    storeProperties.getProperties().putAll(exportStoreProperties.getProperties());
-                }
-            }
+        rtn = new Builder()
+                .graphId(exportGraphId)
+                .library(graphLibrary)
+                .addSchema(schema)
+                .storeProperties(storeProperties)
+                .build();
+        return rtn;
+    }
+
+    private StoreProperties resolveStoreProperties(final StoreProperties exportStoreProperties, final String exportParentStorePropertiesId, final GraphLibrary graphLibrary, final StoreProperties storeStoreProperties) {
+        StoreProperties storeProperties = null;
```

Comment:
If we are using rtn as the returned object above can we stick with it here? Also applies to resoveSchema below.